--- a/build/rust/rust_bindgen_generator.gni
+++ b/build/rust/rust_bindgen_generator.gni
@@ -18,7 +18,7 @@
 
 _rustfmt_path = "${rust_bindgen_root}/bin/rustfmt"
 if (!use_chromium_rust_toolchain &&
-    (host_cpu == "ppc64" || host_cpu == "s390x")) {
+    (host_cpu == "s390x")) {
   _rustfmt_path = rust_sysroot_absolute + "/bin/rustfmt"
 } else if (host_os == "win") {
   _rustfmt_path = "${_rustfmt_path}.exe"
@@ -28,7 +28,7 @@
 # ../lib.
 _libclang_path = clang_base_path
 if (!use_chromium_rust_toolchain &&
-    (host_cpu == "ppc64" || host_cpu == "s390x")) {
+    (host_cpu == "s390x")) {
   _libclang_path = rust_sysroot_absolute + "/lib64"
 } else if (host_os == "win") {
   _libclang_path += "/bin"
