author: Andres Salomon <dilinger@debian.org>
desc: trixie's older gn lacks len().

--- a/base/BUILD.gn
+++ b/base/BUILD.gn
@@ -2882,8 +2882,7 @@ if (is_mac) {
     header = "app_plist_size_buildflags.h"
     header_dir = "base/apple"
 
-    plist_file = read_file("//chrome/app/app-Info.plist", "string")
-    flags = [ "APP_INFO_PLIST_FILE_SIZE=" + len(plist_file) ]
+    flags = [ "APP_INFO_PLIST_FILE_SIZE=" + "unknown" ]
   }
 }
 
--- a/build/rust/rust_bindgen.gni
+++ b/build/rust/rust_bindgen.gni
@@ -112,7 +112,6 @@ template("rust_bindgen") {
   # Calculate `_generated_bindings_rs`.
   _bindgen_outputs = get_target_outputs(":$_rust_bindgen_generator_target_name")
   _bindgen_rs_outputs = filter_include(_bindgen_outputs, [ "*.rs" ])
-  assert(len(_bindgen_rs_outputs) == 1)
   _generated_bindings_rs = _bindgen_rs_outputs[0]
 
   rust_static_library(target_name) {
