--- a/ui/webui/resources/tools/eslint/web_component_missing_deps.ts
+++ b/ui/webui/resources/tools/eslint/web_component_missing_deps.ts
@@ -135,8 +135,8 @@ export const webComponentMissingDeps = E
         for (const tagName of tagNames) {
           const expectedImportedFilenames = getExpectedImportFilenames(tagName);
 
-          if (importedFilenames.intersection(expectedImportedFilenames).size >
-              0) {
+          // Set.intersection is only available in node 22 and higher
+          if (new Set([...importedFilenames].filter((i) => expectedImportedFilenames.has(i))).size > 0) {
             continue;
           }
 
@@ -153,8 +153,7 @@ export const webComponentMissingDeps = E
           }
 
           if (lazyImportedFileNames !== null &&
-              lazyImportedFileNames.intersection(expectedImportedFilenames)
-                      .size > 0) {
+            (new Set([...lazyImportedFileNames].filter((i) => expectedImportedFilenames.has(i))).size > 0)) {
             // Handle case where the dependency is imported lazily (meaning in
             // lazy_load.ts).
             continue;
