--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1687,8 +1687,6 @@ config("no_chromium_code") {
     ]
     if (!is_nacl && (current_toolchain == host_toolchain || !use_xcode_clang)) {
       cflags += [
-        # Disabled for similar reasons as -Wunused-variable.
-        "-Wno-unused-but-set-variable",
 
         # TODO(https://crbug.com/1202159): Clean up and enable.
         "-Wno-misleading-indentation",
