Index: chromium-140.0.7339.41/v8/gni/v8.gni
===================================================================
--- chromium-140.0.7339.41.orig/v8/gni/v8.gni
+++ chromium-140.0.7339.41/v8/gni/v8.gni
@@ -274,7 +274,7 @@ assert(!(v8_enable_webassembly && v8_ena
 if (v8_enable_pointer_compression == "") {
   v8_enable_pointer_compression =
       v8_current_cpu == "arm64" || v8_current_cpu == "x64" ||
-      v8_current_cpu == "loong64"
+      v8_current_cpu == "loong64" || v8_current_cpu == "ppc64"
 }
 
 # The Wasm interpreter is currently supported only on arm64 and x64, on
