Index: chromium-143.0.7499.40/v8/gni/v8.gni
===================================================================
--- chromium-143.0.7499.40.orig/v8/gni/v8.gni
+++ chromium-143.0.7499.40/v8/gni/v8.gni
@@ -298,7 +298,7 @@ assert(
 if (v8_enable_pointer_compression == "") {
   v8_enable_pointer_compression =
       v8_current_cpu == "arm64" || v8_current_cpu == "x64" ||
-      v8_current_cpu == "loong64"
+      v8_current_cpu == "loong64" || v8_current_cpu == "ppc64"
 }
 
 # The Wasm interpreter is currently supported only on arm64 and x64, on
