Index: chromium-141.0.7390.37/v8/gni/v8.gni
===================================================================
--- chromium-141.0.7390.37.orig/v8/gni/v8.gni
+++ chromium-141.0.7390.37/v8/gni/v8.gni
@@ -281,7 +281,7 @@ assert(
 if (v8_enable_pointer_compression == "") {
   v8_enable_pointer_compression =
       v8_current_cpu == "arm64" || v8_current_cpu == "x64" ||
-      v8_current_cpu == "loong64"
+      v8_current_cpu == "loong64" || v8_current_cpu == "ppc64"
 }
 
 # The Wasm interpreter is currently supported only on arm64 and x64, on
