work around unnecessary qualification build error. Probably libstdc++ specific.

--- a/third_party/ink/src/ink/strokes/internal/noise_generator.cc
+++ b/third_party/ink/src/ink/strokes/internal/noise_generator.cc
@@ -66,7 +66,7 @@ void NoiseGenerator::AdvanceInputBy(floa
 
     // Set `progress_` equal to its fractional part (i.e. `progress_` mod 1).
     float unused;
-    progress_ = std::modff(progress_, &unused);
+    progress_ = modff(progress_, &unused);
   }
 }
 
