author: Andres Salomon <dilinger@debian.org>

more base::StringPiece -> std::string conversion headaches..

--- a/net/dns/host_resolver_cache.h
+++ b/net/dns/host_resolver_cache.h
@@ -144,12 +144,14 @@ class NET_EXPORT HostResolverCache final
     }
 
     bool operator()(const Key& lhs, const KeyRef& rhs) const {
+      const std::string dn{rhs.domain_name};
       return std::tie(lhs.domain_name, lhs.network_anonymization_key) <
-             std::tie(rhs.domain_name, *rhs.network_anonymization_key);
+             std::tie(dn, *rhs.network_anonymization_key);
     }
 
     bool operator()(const KeyRef& lhs, const Key& rhs) const {
-      return std::tie(lhs.domain_name, *lhs.network_anonymization_key) <
+      const std::string dn{lhs.domain_name};
+      return std::tie(dn, *lhs.network_anonymization_key) <
              std::tie(rhs.domain_name, rhs.network_anonymization_key);
     }
   };
