--- a/build/rust/gni_impl/run_bindgen.py
+++ b/build/rust/gni_impl/run_bindgen.py
@@ -66,6 +66,28 @@ def main():
       nargs="*")
   args = parser.parse_args()
 
+  # Debian pre-generated output
+  deb_want_pre_gen = False
+  if args.output:
+    import shutil
+    assert args.output.endswith(".rs")
+    deb_bg_dir = "../../debian/bindgen"
+    deb_pre_gen = deb_bg_dir + "/" + ".".join(args.output.split("/")[-2:])
+    deb_pre_gen_dep = deb_pre_gen[:-3] + ".d"
+    deb_pre_gen_wrap = deb_pre_gen[:-3] + ".wrap.c"
+    if not os.path.isdir(deb_bg_dir):
+      pass
+    elif not os.path.exists(args.bindgen_exe):
+      print(f"note: using pre-generated output from {deb_pre_gen} et al.", file=sys.stderr)
+      shutil.copyfile(deb_pre_gen, args.output)
+      if (args.depfile):
+        shutil.copyfile(deb_pre_gen_dep, args.depfile)
+      if (args.wrap_static_fns):
+        shutil.copyfile(deb_pre_gen_wrap, args.wrap_static_fns)
+      sys.exit(0)
+    elif not os.path.exists(deb_pre_gen):
+      deb_want_pre_gen = True
+
   # Abort if `TARGET` exists in the environment. Cargo sets `TARGET` when
   # running build scripts and bindgen will try to be helpful by using that value
   # if it's set. In practice we've seen a case where someone had the value set
@@ -139,6 +161,22 @@ def main():
         pass
       raise
 
+  # Debian pre-generated output
+  if deb_want_pre_gen:
+    print(f"note: saving generated output to {deb_pre_gen} et al.", file=sys.stderr)
+    shutil.copyfile(args.output, deb_pre_gen)
+    if (args.wrap_static_fns):
+      shutil.copyfile(args.wrap_static_fns, deb_pre_gen_wrap)
+    if (args.depfile):
+      # Filter out system paths from the depfile
+      with open(args.depfile, "r") as f:
+        dep_line = f.read()
+      assert "\n" not in dep_line
+      deps = dep_line.split()
+      deps_filt = [d for d in deps if not (d.startswith("/usr/include/") or "/usr/lib/llvm-" in d)]
+      with open(deb_pre_gen_dep, "w") as f:
+        f.write(" ".join(deps_filt))
+
 
 if __name__ == '__main__':
   main()
--- a/build/rust/rust_bindgen_generator.gni
+++ b/build/rust/rust_bindgen_generator.gni
@@ -147,8 +147,6 @@ template("rust_bindgen_generator") {
 
     script = rebase_path("//build/rust/gni_impl/run_bindgen.py")
     inputs = [
-      _bindgen_path,
-      _rustfmt_path,
       "//build/action_helpers.py",
       "//build/gn_helpers.py",
       "//build/rust/gni_impl/filter_clang_args.py",
