description: disable dependency on chrome/android
author: Michael Gilbert <mgilbert@debian.org>

--- a/BUILD.gn
+++ b/BUILD.gn
@@ -351,7 +351,7 @@ group("gn_all") {
     # Allow building md5sum tool for other OSes (android is added below).
     # For Windows, see: crbug.com/1090428.
     # TODO(cavalcantii): move it out of 'android' folder.
-    if (!is_win) {
+    if (is_android) {
       deps += [ "//tools/android/md5sum" ]
     }
 
--- a/chrome/common/profiler/unwind_util.cc
+++ b/chrome/common/profiler/unwind_util.cc
@@ -20,7 +20,6 @@
 #include "build/branding_buildflags.h"
 #include "build/build_config.h"
 #include "chrome/common/channel_info.h"
-#include "chrome/common/profiler/native_unwinder_android_map_delegate_impl.h"
 #include "chrome/common/profiler/process_type.h"
 #include "components/metrics/call_stack_profile_params.h"
 #include "components/version_info/channel.h"
