description: Disable ANGLE perftests
author: Michel Le Bihan <michel@lebihan.pl>

Index: chromium-104.0.5112.65/chrome/test/BUILD.gn
===================================================================
--- chromium-104.0.5112.65.orig/chrome/test/BUILD.gn
+++ chromium-104.0.5112.65/chrome/test/BUILD.gn
@@ -4859,7 +4859,6 @@ template("performance_test_suite_templat
     }
 
     if (!is_chromeos_lacros) {
-      data_deps += [ "//third_party/angle/src/tests:angle_perftests" ]
     }
 
     if (!is_android && !is_chromeos_lacros) {
Index: chromium-104.0.5112.65/third_party/angle/src/tests/BUILD.gn
===================================================================
--- chromium-104.0.5112.65.orig/third_party/angle/src/tests/BUILD.gn
+++ chromium-104.0.5112.65/third_party/angle/src/tests/BUILD.gn
@@ -1623,12 +1580,11 @@ group("angle_tests") {
   testonly = true
   deps = [
     ":angle_end2end_tests",
-    ":angle_perftests",
     ":angle_system_info_test",
     ":angle_unittests",
     "capture_replay_tests",
   ]
-  if (!is_fuchsia) {
+  if (false) {
     deps += [
       ":angle_white_box_perftests",
       ":angle_white_box_tests",
Index: chromium-104.0.5112.65/third_party/angle/gni/angle.gni
===================================================================
--- chromium-104.0.5112.65.orig/third_party/angle/gni/angle.gni
+++ chromium-104.0.5112.65/third_party/angle/gni/angle.gni
@@ -249,7 +249,7 @@ declare_args() {
       is_apple &&
       (angle_enable_glsl || angle_enable_metal || use_fuzzing_engine)
 
-  angle_has_histograms = angle_has_build
+  angle_has_histograms = false
 
   # True if we want to build the ANGLE test suites. On by default in most configs.
   angle_build_tests = angle_has_build
