description: disable building font tests
author: Michael Gilbert <mgilbert@debian.org>

--- a/skia/BUILD.gn
+++ b/skia/BUILD.gn
@@ -863,7 +863,7 @@ group("test_fonts_resources") {
   if (is_apple) {
     deps += [ ":test_fonts_bundle_data" ]
     data_deps += [ ":test_fonts_bundle_data" ]
-  } else {
+  } else if (false) {
     deps += [ "//third_party/test_fonts" ]
     data_deps += [ "//third_party/test_fonts" ]
   }
--- a/base/test/BUILD.gn
+++ b/base/test/BUILD.gn
@@ -198,7 +198,6 @@ static_library("test_support") {
 
   if (is_linux || is_chromeos) {
     sources += [ "test_file_util_linux.cc" ]
-    public_deps += [ "//third_party/test_fonts/fontconfig:test_support" ]
   }
 
   if (is_mac) {
