description: Disable Widevine CDM component updater
author: Michel Le Bihan <michel@lebihan.pl>

--- a/chrome/browser/component_updater/registration.cc
+++ b/chrome/browser/component_updater/registration.cc
@@ -130,7 +130,7 @@ void RegisterComponentsForUpdate() {
 #endif  // BUILDFLAG(IS_MAC)
 
 #if BUILDFLAG(ENABLE_MEDIA_FOUNDATION_WIDEVINE_CDM)
-  RegisterMediaFoundationWidevineCdmComponent(cus);
+//  RegisterMediaFoundationWidevineCdmComponent(cus);
 #endif
 
 #if BUILDFLAG(ENABLE_WIDEVINE_CDM_COMPONENT)
