Index: chromium-119.0.6045.59/third_party/sqlite/src/amalgamation/sqlite3.c
===================================================================
--- chromium-119.0.6045.59.orig/third_party/sqlite/src/amalgamation/sqlite3.c
+++ chromium-119.0.6045.59/third_party/sqlite/src/amalgamation/sqlite3.c
@@ -14576,7 +14576,8 @@ typedef INT16_TYPE LogEst;
 # if defined(i386)      || defined(__i386__)      || defined(_M_IX86) ||    \
      defined(__x86_64)  || defined(__x86_64__)    || defined(_M_X64)  ||    \
      defined(_M_AMD64)  || defined(_M_ARM)        || defined(__x86)   ||    \
-     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64)
+     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64) ||   \
+     defined(__powerpc64__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
 #   define SQLITE_BYTEORDER    1234
 # elif defined(sparc)     || defined(__ppc__) || \
        defined(__ARMEB__) || defined(__AARCH64EB__)
Index: chromium-119.0.6045.59/third_party/sqlite/src/amalgamation_dev/sqlite3.c
===================================================================
--- chromium-119.0.6045.59.orig/third_party/sqlite/src/amalgamation_dev/sqlite3.c
+++ chromium-119.0.6045.59/third_party/sqlite/src/amalgamation_dev/sqlite3.c
@@ -14589,7 +14589,8 @@ typedef INT16_TYPE LogEst;
 # if defined(i386)      || defined(__i386__)      || defined(_M_IX86) ||    \
      defined(__x86_64)  || defined(__x86_64__)    || defined(_M_X64)  ||    \
      defined(_M_AMD64)  || defined(_M_ARM)        || defined(__x86)   ||    \
-     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64)
+     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64) ||   \
+     defined(__powerpc64__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
 #   define SQLITE_BYTEORDER    1234
 # elif defined(sparc)     || defined(__ppc__) || \
        defined(__ARMEB__) || defined(__AARCH64EB__)
Index: chromium-119.0.6045.59/third_party/sqlite/src/ext/rtree/rtree.c
===================================================================
--- chromium-119.0.6045.59.orig/third_party/sqlite/src/ext/rtree/rtree.c
+++ chromium-119.0.6045.59/third_party/sqlite/src/ext/rtree/rtree.c
@@ -474,7 +474,8 @@ struct RtreeMatchArg {
 # if defined(i386)      || defined(__i386__)      || defined(_M_IX86) ||    \
      defined(__x86_64)  || defined(__x86_64__)    || defined(_M_X64)  ||    \
      defined(_M_AMD64)  || defined(_M_ARM)        || defined(__x86)   ||    \
-     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64)
+     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64) ||  \
+     defined(__powerpc64__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
 #   define SQLITE_BYTEORDER    1234
 # elif defined(sparc)     || defined(__ppc__) || \
        defined(__ARMEB__) || defined(__AARCH64EB__)
Index: chromium-119.0.6045.59/third_party/sqlite/src/src/sqliteInt.h
===================================================================
--- chromium-119.0.6045.59.orig/third_party/sqlite/src/src/sqliteInt.h
+++ chromium-119.0.6045.59/third_party/sqlite/src/src/sqliteInt.h
@@ -905,7 +905,8 @@ typedef INT16_TYPE LogEst;
 # if defined(i386)      || defined(__i386__)      || defined(_M_IX86) ||    \
      defined(__x86_64)  || defined(__x86_64__)    || defined(_M_X64)  ||    \
      defined(_M_AMD64)  || defined(_M_ARM)        || defined(__x86)   ||    \
-     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64)
+     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64) ||   \
+     defined(__powerpc64__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
 #   define SQLITE_BYTEORDER    1234
 # elif defined(sparc)     || defined(__ppc__) || \
        defined(__ARMEB__) || defined(__AARCH64EB__)
