Index: chromium-120.0.6099.71/third_party/sqlite/src/amalgamation/sqlite3.c
===================================================================
--- chromium-120.0.6099.71.orig/third_party/sqlite/src/amalgamation/sqlite3.c
+++ chromium-120.0.6099.71/third_party/sqlite/src/amalgamation/sqlite3.c
@@ -14656,7 +14656,8 @@ typedef INT16_TYPE LogEst;
 # if defined(i386)      || defined(__i386__)      || defined(_M_IX86) ||    \
      defined(__x86_64)  || defined(__x86_64__)    || defined(_M_X64)  ||    \
      defined(_M_AMD64)  || defined(_M_ARM)        || defined(__x86)   ||    \
-     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64)
+     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64) ||   \
+     defined(__powerpc64__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
 #   define SQLITE_BYTEORDER    1234
 # elif defined(sparc)     || defined(__ppc__) || \
        defined(__ARMEB__) || defined(__AARCH64EB__)
Index: chromium-120.0.6099.71/third_party/sqlite/src/amalgamation_dev/sqlite3.c
===================================================================
--- chromium-120.0.6099.71.orig/third_party/sqlite/src/amalgamation_dev/sqlite3.c
+++ chromium-120.0.6099.71/third_party/sqlite/src/amalgamation_dev/sqlite3.c
@@ -14669,7 +14669,8 @@ typedef INT16_TYPE LogEst;
 # if defined(i386)      || defined(__i386__)      || defined(_M_IX86) ||    \
      defined(__x86_64)  || defined(__x86_64__)    || defined(_M_X64)  ||    \
      defined(_M_AMD64)  || defined(_M_ARM)        || defined(__x86)   ||    \
-     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64)
+     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64) ||   \
+     defined(__powerpc64__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
 #   define SQLITE_BYTEORDER    1234
 # elif defined(sparc)     || defined(__ppc__) || \
        defined(__ARMEB__) || defined(__AARCH64EB__)
Index: chromium-120.0.6099.71/third_party/sqlite/src/ext/rtree/rtree.c
===================================================================
--- chromium-120.0.6099.71.orig/third_party/sqlite/src/ext/rtree/rtree.c
+++ chromium-120.0.6099.71/third_party/sqlite/src/ext/rtree/rtree.c
@@ -479,7 +479,8 @@ struct RtreeMatchArg {
 # if defined(i386)      || defined(__i386__)      || defined(_M_IX86) ||    \
      defined(__x86_64)  || defined(__x86_64__)    || defined(_M_X64)  ||    \
      defined(_M_AMD64)  || defined(_M_ARM)        || defined(__x86)   ||    \
-     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64)
+     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64) ||  \
+     defined(__powerpc64__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
 #   define SQLITE_BYTEORDER    1234
 # elif defined(sparc)     || defined(__ppc__) || \
        defined(__ARMEB__) || defined(__AARCH64EB__)
Index: chromium-120.0.6099.71/third_party/sqlite/src/src/sqliteInt.h
===================================================================
--- chromium-120.0.6099.71.orig/third_party/sqlite/src/src/sqliteInt.h
+++ chromium-120.0.6099.71/third_party/sqlite/src/src/sqliteInt.h
@@ -928,7 +928,8 @@ typedef INT16_TYPE LogEst;
 # if defined(i386)      || defined(__i386__)      || defined(_M_IX86) ||    \
      defined(__x86_64)  || defined(__x86_64__)    || defined(_M_X64)  ||    \
      defined(_M_AMD64)  || defined(_M_ARM)        || defined(__x86)   ||    \
-     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64)
+     defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64) ||   \
+     defined(__powerpc64__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
 #   define SQLITE_BYTEORDER    1234
 # elif defined(sparc)     || defined(__ppc__) || \
        defined(__ARMEB__) || defined(__AARCH64EB__)
