Index: chromium-141.0.7390.37/third_party/angle/src/libANGLE/Constants.h
===================================================================
--- chromium-141.0.7390.37.orig/third_party/angle/src/libANGLE/Constants.h
+++ chromium-141.0.7390.37/third_party/angle/src/libANGLE/Constants.h
@@ -9,6 +9,7 @@
 #ifndef LIBANGLE_CONSTANTS_H_
 #define LIBANGLE_CONSTANTS_H_
 
+#include <cstddef>
 #include "common/platform.h"
 
 #include <stddef.h>
