Index: chromium-127.0.6533.72/third_party/dawn/src/dawn/common/Platform.h
===================================================================
--- chromium-127.0.6533.72.orig/third_party/dawn/src/dawn/common/Platform.h
+++ chromium-127.0.6533.72/third_party/dawn/src/dawn/common/Platform.h
@@ -153,9 +153,9 @@
 #define DAWN_PLATFORM_IS_MIPS64 1
 #endif
 
-#elif defiend(__s390__)
+#elif defined(__s390__)
 #define DAWN_PLATFORM_IS_S390 1
-#elif defiend(__s390x__)
+#elif defined(__s390x__)
 #define DAWN_PLATFORM_IS_S390X 1
 
 #elif defined(__PPC__)
Index: chromium-127.0.6533.72/third_party/dawn/src/dawn/common/Assert.cpp
===================================================================
--- chromium-127.0.6533.72.orig/third_party/dawn/src/dawn/common/Assert.cpp
+++ chromium-127.0.6533.72/third_party/dawn/src/dawn/common/Assert.cpp
@@ -52,9 +52,9 @@ void BreakPoint() {
     __asm__ __volatile__("ebreak");
 #elif DAWN_PLATFORM_IS(MIPS)
     __asm__ __volatile__("break");
-#elif DAWN_PLATFORM_IS(S390) || DAWN_PLATFORM_IS_(S390X)
+#elif DAWN_PLATFORM_IS(S390) || DAWN_PLATFORM_IS(S390X)
     __asm__ __volatile__(".word 0x0001");
-#elif DAWN_PLATFORM_IS(PPC) || DAWN_PLATFORM_IS_(PPC64)
+#elif DAWN_PLATFORM_IS(PPC) || DAWN_PLATFORM_IS(PPC64)
     __asm__ __volatile__("twge 2,2");
 #else
 #error "Unsupported platform"
