Index: chromium-134.0.6998.35/base/allocator/partition_allocator/src/partition_alloc/page_allocator_constants.h
===================================================================
--- chromium-134.0.6998.35.orig/base/allocator/partition_allocator/src/partition_alloc/page_allocator_constants.h
+++ chromium-134.0.6998.35/base/allocator/partition_allocator/src/partition_alloc/page_allocator_constants.h
@@ -209,7 +209,11 @@ SystemPageBaseMask() {
   return ~SystemPageOffsetMask();
 }
 
+#if PA_BUILDFLAG(PA_ARCH_CPU_PPC64_FAMILY)
+constexpr size_t kPageMetadataShift = 6;  // 64 bytes per partition page.
+#else
 constexpr size_t kPageMetadataShift = 5;  // 32 bytes per partition page.
+#endif
 constexpr size_t kPageMetadataSize = 1 << kPageMetadataShift;
 
 }  // namespace internal
