--- a/media/capture/capture_switches.cc
+++ b/media/capture/capture_switches.cc
@@ -48,7 +48,7 @@ BASE_FEATURE(TabCaptureInfobarLinks, bas
 #if defined(WEBRTC_USE_PIPEWIRE)
 // Controls whether the PipeWire support for cameras is enabled on the
 // Wayland display server.
-BASE_FEATURE(WebRtcPipeWireCamera, base::FEATURE_DISABLED_BY_DEFAULT);
+BASE_FEATURE(WebRtcPipeWireCamera, base::FEATURE_ENABLED_BY_DEFAULT);
 #endif  // defined(WEBRTC_USE_PIPEWIRE)
 
 #if BUILDFLAG(IS_WIN)
