--- a/ui/gfx/linux/drm_util_linux.cc
+++ b/ui/gfx/linux/drm_util_linux.cc
@@ -40,6 +40,7 @@ int GetFourCCFormatFromBufferFormat(gfx:
       return DRM_FORMAT_ABGR2101010;
     case gfx::BufferFormat::RGBA_F16:
       return DRM_FORMAT_ABGR16161616F;
+#if 0
     case gfx::BufferFormat::YVU_420:
       return DRM_FORMAT_YVU420;
     case gfx::BufferFormat::YUV_420_BIPLANAR:
@@ -48,6 +49,13 @@ int GetFourCCFormatFromBufferFormat(gfx:
       return DRM_FORMAT_INVALID;
     case gfx::BufferFormat::P010:
       return DRM_FORMAT_P010;
+#else
+    case gfx::BufferFormat::YVU_420:
+    case gfx::BufferFormat::YUV_420_BIPLANAR:
+    case gfx::BufferFormat::YUVA_420_TRIPLANAR:
+    case gfx::BufferFormat::P010:
+      return DRM_FORMAT_INVALID;
+#endif
   }
   return DRM_FORMAT_INVALID;
 }
