--- a/media/gpu/v4l2/legacy/v4l2_video_decoder_backend_stateful.cc
+++ b/media/gpu/v4l2/legacy/v4l2_video_decoder_backend_stateful.cc
@@ -319,6 +319,8 @@ void V4L2StatefulVideoDecoderBackend::On
 
   // Try to progress on our work since we may have dequeued input buffers.
   DoDecodeWork();
+
+  device_->SchedulePri();
 }
 
 void V4L2StatefulVideoDecoderBackend::EnqueueOutputBuffers() {
@@ -548,6 +550,9 @@ bool V4L2StatefulVideoDecoderBackend::In
     // request will be processed in OnChangeResolutionDone(), when the CAPTURE
     // queue starts streaming.
     DVLOGF(2) << "Flush request to be processed after CAPTURE queue starts";
+    if (!input_queue_->SendStopCommand())
+      VLOGF(1) << "Stop command failed";
+    device_->SchedulePoll();
   }
 
   return true;
--- a/media/gpu/v4l2/v4l2_device.cc
+++ b/media/gpu/v4l2/v4l2_device.cc
@@ -142,6 +142,12 @@ bool V4L2Device::Open(Type type, uint32_
     return false;
   }
 
+  device_poll_prod_fd_.reset(eventfd(0, EFD_NONBLOCK | EFD_CLOEXEC));
+  if (!device_poll_prod_fd_.is_valid()) {
+    VLOGF(1) << "Failed creating a poll interrupt fd";
+    return false;
+  }
+
   return true;
 }
 
@@ -348,28 +354,55 @@ int V4L2Device::Ioctl(int request, void*
 }
 
 bool V4L2Device::Poll(bool poll_device, bool* event_pending) {
-  std::array<struct pollfd, 2> pollfds;
-  nfds_t nfds;
-  int pollfd = -1;
-
-  pollfds[0].fd = device_poll_interrupt_fd_.get();
-  pollfds[0].events = POLLIN | POLLERR;
-  nfds = 1;
-
-  if (poll_device) {
-    DVLOGF(5) << "adding device fd to poll() set";
-    pollfds[nfds].fd = device_fd_.get();
-    pollfds[nfds].events = POLLIN | POLLOUT | POLLERR | POLLPRI;
-    pollfd = nfds;
+  while (true) {
+    struct pollfd pollfds[3] = { { 0 } };
+    nfds_t nfds;
+    int pollfd = -1;
+    int prod_n;
+    const unsigned int cur_events = events_.load();
+    const unsigned int poll_events = cur_events ? cur_events : POLLIN | POLLOUT | POLLERR | POLLPRI;
+
+    pollfds[0].fd = device_poll_interrupt_fd_.get();
+    pollfds[0].events = POLLIN | POLLERR;
+    nfds = 1;
+
+    pollfds[nfds].fd = device_poll_prod_fd_.get();
+    pollfds[nfds].events = POLLIN | POLLERR;
+    prod_n = nfds;
     nfds++;
-  }
 
-  if (HANDLE_EINTR(poll(pollfds.data(), nfds, -1)) == -1) {
-    VPLOGF(1) << "poll() failed";
-    return false;
+    if (poll_device) {
+      DVLOGF(5) << "adding device fd to poll() set";
+      pollfds[nfds].fd = device_fd_.get();
+      pollfds[nfds].events = poll_events;
+      pollfd = nfds;
+      nfds++;
+    }
+
+    DVLOGF(5) << "pollfd-pre " << pollfd << " events 0x" << std::hex << pollfds[pollfd].events;
+
+    if (HANDLE_EINTR(poll(pollfds, nfds, -1)) == -1) {
+      VPLOGF(1) << "poll() failed";
+      return false;
+    }
+
+    DVLOGF(5) << "pollfd-post " << pollfd << " revents 0x" << std::hex << pollfds[pollfd].revents << " events 0x" << std::hex << pollfds[pollfd].events;
+
+    if (pollfds[prod_n].revents & POLLIN) {
+      uint64_t n;
+      if (read(device_poll_prod_fd_.get(), &n, sizeof(n)) != sizeof(n))
+        VPLOGF(1) << "Prod read at best confused";
+      continue;
+    }
+
+    if (poll_events == POLLPRI && pollfds[0].revents == 0 &&
+        (pollfd != -1 && !(pollfds[pollfd].revents & POLLPRI)))
+      continue;
+
+    events_.store(0);
+    *event_pending = (pollfd != -1 && pollfds[pollfd].revents & POLLPRI);
+    return true;
   }
-  *event_pending = (pollfd != -1 && pollfds[pollfd].revents & POLLPRI);
-  return true;
 }
 
 void* V4L2Device::Mmap(void* addr,
@@ -397,6 +430,19 @@ bool V4L2Device::SetDevicePollInterrupt(
   return true;
 }
 
+bool V4L2Device::PollProd() {
+  DVLOGF(5);
+
+  const uint64_t buf = 1;
+  if (HANDLE_EINTR(write(device_poll_prod_fd_.get(), &buf, sizeof(buf))) ==
+      -1) {
+    VPLOGF(1) << "write() failed";
+    return false;
+  }
+  return true;
+}
+
+
 bool V4L2Device::ClearDevicePollInterrupt() {
   DVLOGF(5);
 
@@ -657,9 +703,31 @@ bool V4L2Device::StopPolling() {
 void V4L2Device::SchedulePoll() {
   DCHECK_CALLED_ON_VALID_SEQUENCE(client_sequence_checker_);
 
+  if (!device_poller_)
+    return;
+
+  LOG(INFO) << __func__ << ": events " << std::hex << events_.load();
+
+  const unsigned int prev = events_.fetch_or(POLLIN | POLLOUT);
+  if ((prev & (POLLIN | POLLOUT)) != (POLLIN | POLLOUT)) {
+    PollProd();
+  }
+
+  if (!device_poller_->IsPolling())
+    return;
+
+  device_poller_->SchedulePoll();
+}
+
+void V4L2Device::SchedulePri() {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(client_sequence_checker_);
+
   if (!device_poller_ || !device_poller_->IsPolling())
     return;
 
+  LOG(INFO) << __func__ << ": events " << std::hex << events_.load();
+
+  events_.fetch_or(POLLPRI);
   device_poller_->SchedulePoll();
 }
 
--- a/media/gpu/v4l2/v4l2_device.h
+++ b/media/gpu/v4l2/v4l2_device.h
@@ -13,6 +13,7 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <atomic>
 #include <optional>
 #include <vector>
 
@@ -128,6 +129,8 @@ class MEDIA_GPU_EXPORT V4L2Device
   bool SetDevicePollInterrupt();
   bool ClearDevicePollInterrupt();
 
+  bool PollProd();
+
   // Wrappers for standard mmap/munmap system calls.
   void* Mmap(void* addr,
              unsigned int len,
@@ -186,6 +189,7 @@ class MEDIA_GPU_EXPORT V4L2Device
   // Schedule a polling event if polling is enabled. This method is intended
   // to be called from V4L2Queue, clients should not need to call it directly.
   void SchedulePoll();
+  void SchedulePri();
 
   // Attempt to dequeue a V4L2 event and return it.
   std::optional<struct v4l2_event> DequeueEvent();
@@ -275,6 +279,9 @@ class MEDIA_GPU_EXPORT V4L2Device
   // interrupted.
   base::ScopedFD device_poll_interrupt_fd_;
 
+  std::atomic<unsigned int> events_{0};
+  base::ScopedFD device_poll_prod_fd_;
+
   // Associates a v4l2_buf_type to its queue.
   base::flat_map<enum v4l2_buf_type, V4L2Queue*> queues_;
 
--- a/media/gpu/v4l2/v4l2_queue.cc
+++ b/media/gpu/v4l2/v4l2_queue.cc
@@ -1454,11 +1454,15 @@ std::pair<bool, V4L2ReadableBufferRef> V
     // should not call this method after the last buffer is dequeued.
     switch (errno) {
       case EAGAIN:
-      case EPIPE:
         // This is not an error so we'll need to continue polling but won't
         // provide a buffer.
+        DVLOGF(1) << "Polling";
         schedule_poll_cb_.Run();
         return std::make_pair(true, nullptr);
+      case EPIPE:
+        // Event or similar
+        DVLOGF(1) << "Pipe";
+        return std::make_pair(true, nullptr);
       default:
         VPQLOGF(1) << "VIDIOC_DQBUF failed";
         return std::make_pair(false, nullptr);
