--- a/ui/ozone/platform/wayland/gpu/wayland_surface_factory.cc
+++ b/ui/ozone/platform/wayland/gpu/wayland_surface_factory.cc
@@ -249,20 +249,17 @@ scoped_refptr<gfx::NativePixmap> Wayland
     return nullptr;
   }
 #if defined(WAYLAND_GBM)
-  auto* gbm_device = buffer_manager_->GetGbmDevice();
-  if (gbm_device && gbm_device->CanCreateBufferForFormat(
-                        GetFourCCFormatFromBufferFormat(format))) {
-    scoped_refptr<GbmPixmapWayland> pixmap =
-        base::MakeRefCounted<GbmPixmapWayland>(buffer_manager_);
+  scoped_refptr<GbmPixmapWayland> pixmap =
+      base::MakeRefCounted<GbmPixmapWayland>(buffer_manager_);
 
-    if (!pixmap->InitializeBuffer(widget, size, format, usage,
-                                  framebuffer_size)) {
-      return nullptr;
-    }
-    return pixmap;
+  if (!pixmap->InitializeBuffer(widget, size, format, usage,
+                                framebuffer_size)) {
+    return nullptr;
   }
-#endif
+  return pixmap;
+#else
   return nullptr;
+#endif
 }
 
 void WaylandSurfaceFactory::CreateNativePixmapAsync(
--- a/ui/ozone/platform/wayland/ozone_platform_wayland.cc
+++ b/ui/ozone/platform/wayland/ozone_platform_wayland.cc
@@ -71,8 +71,6 @@
 #endif
 
 #if defined(WAYLAND_GBM)
-#include "ui/gfx/linux/drm_util_linux.h"
-#include "ui/gfx/linux/gbm_device.h"
 #include "ui/ozone/platform/wayland/common/drm_render_node_path_finder.h"
 #endif
 
@@ -200,15 +198,6 @@ class OzonePlatformWayland : public Ozon
       if (!buffer_manager_->SupportsFormat(format)) {
         return false;
       }
-      // Return false here if creating buffers for certain formats is not
-      // possible (e.g. YUV formats are not supported by linux system libgbm
-      // gbm_bo_create) even though |buffer_manager_| may indicate it can be
-      // imported as wl_buffer.
-      auto* gbm_device = buffer_manager_->GetGbmDevice();
-      if (!gbm_device || !gbm_device->CanCreateBufferForFormat(
-                             GetFourCCFormatFromBufferFormat(format))) {
-        return false;
-      }
     } else {
       if (supported_buffer_formats_.find(format) ==
           supported_buffer_formats_.end()) {
