From 56a580d2c91ff870bb0f6a5ab2e18d8ba2f47c1e Mon Sep 17 00:00:00 2001
From: John Cox <jc@kynesim.co.uk>
Date: Mon, 20 May 2024 17:40:58 +0100
Subject: [PATCH 12/17] Init: Prefer GLES to everything else on init

---
 ui/gl/init/gl_factory.cc | 9 +++++++++
 1 file changed, 9 insertions(+)

--- a/src/ui/gl/init/gl_factory.cc
+++ b/src/ui/gl/init/gl_factory.cc
@@ -103,6 +103,15 @@ GLImplementationParts GetRequestedGLImpl
   std::optional<GLImplementationParts> impl_from_cmdline =
       GetRequestedGLImplementationFromCommandLine(cmd, fallback_to_software_gl);
 
+  // RPI: We always want EGL as our preferred render
+  // Easiest to simply add here - angle + GLES +/- EGL crashes
+  // so add here after the angle add.
+  //
+  // BEWARE: falling back from EGL to anything else with --in-process-gpu
+  // set produces a segfault when we exit so ideally we would only do this
+  // on Pi4/(F)KMS?
+  allowed_impls.insert(allowed_impls.begin(), GLImplementationParts(kGLImplementationEGLGLES2));
+
   // The default implementation is always the first one in list.
   if (!impl_from_cmdline)
     return allowed_impls[0];
