--- a/src/ui/gtk/gtk_color_mixers.cc
+++ b/src/ui/gtk/gtk_color_mixers.cc
@@ -261,9 +261,9 @@ void AddGtkNativeColorMixer(ui::ColorPro
       GetBorderColor(header_selector +
                      ":backdrop GtkSeparator#separator.vertical.titlebutton")};
   mixer[ui::kColorNativeTabForegroundInactiveFrameActive] = {
-      GetFgColor(header_selector + " GtkLabel#label.title")};
+      label_fg};
   mixer[ui::kColorNativeTabForegroundInactiveFrameInactive] = {
-      GetFgColor(header_selector_inactive + " GtkLabel#label.title")};
+      label_fg};
   mixer[ui::kColorNativeToolbarBackground] = {toolbar_color};
   mixer[ui::kColorNativeTextfieldBorderUnfocused] = {entry_border};
   mixer[ui::kColorNativeButtonBorder] = {button_border};
