--- a/src/pi-util/defargs_arm64-bullseye.gn
+++ b/src/pi-util/defargs_arm64-bullseye.gn
@@ -8,9 +8,6 @@ target_os = "linux"
 # Enables screen sharing in hangouts
 enable_hangout_services_extension = true
 
-ffmpeg_branding = "Chrome"
-proprietary_codecs = true
-
 enable_widevine = true
 
 # Dawn seems to really want Vulkan - Chrome doesn't like our Vulkan
--- a/src/pi-util/defargs_armv7-bullseye.gn
+++ b/src/pi-util/defargs_armv7-bullseye.gn
@@ -28,9 +28,6 @@ enable_hangout_services_extension = true
 # enable the validating decode otherwise we get a fatal GPU error
 enable_validating_command_decoder = true
 
-ffmpeg_branding = "Chrome"
-proprietary_codecs = true
-
 disable_fieldtrial_testing_config = true
 enable_nacl = false
 blink_symbol_level = 0
--- a/src/pi-util/gngen.py
+++ b/src/pi-util/gngen.py
@@ -2,7 +2,7 @@
 
 import os, ast, fileinput, subprocess, sys
 
-def docopy(name, vars, is_debug=False, is_ozone=False):
+def docopy(name, vars, is_debug=False, is_ozone=False, extra_codecs=True):
     dir_suffix = ""
     deb_str = "false"
 
@@ -19,6 +19,9 @@ def docopy(name, vars, is_debug=False, i
         deb_str = "false"
         dir_suffix = dir_suffix + "-rel"
 
+    if not extra_codecs:
+        dir_suffix = dir_suffix + "-std"
+
 
     dest_dir = os.path.join("out", name + dir_suffix)
     src_file = os.path.join("pi-util", "defargs_" + name + ".gn")
@@ -37,16 +40,29 @@ def docopy(name, vars, is_debug=False, i
 
     dargs.write('# -- created by ' + sys.argv[0] + '\n')
     dargs.write('is_debug = ' + deb_str + '\n')
+    dargs.write('cc_wrapper = "ccache"\n')
     dargs.write('use_ozone = ' + ozone_str + '\n')
     if is_ozone:
         dargs.write('ozone_platform_x11 = true\n')
         dargs.write('use_v4l2_codec = true\n')
         dargs.write('use_v4l2_codec_rpi = true\n')
+    if extra_codecs:
+        dargs.write('ffmpeg_branding = "Chrome"\n')
+        dargs.write('proprietary_codecs = true\n')
+    else:
+        dargs.write('ffmpeg_branding = "Chromium"\n')
+        dargs.write('proprietary_codecs = false\n')
 
     dargs.write('target_sysroot = "' + vars["target_sysroot"] + '"\n')
-    dargs.write('google_api_key = "' + vars["google_api_key"] + '"\n')
-    dargs.write('google_default_client_id = "' + vars["google_default_client_id"] + '"\n')
-    dargs.write('google_default_client_secret = "' + vars["google_default_client_secret"] + '"\n')
+    #dargs.write('google_api_key = "' + vars["google_api_key"] + '"\n')
+    #dargs.write('google_default_client_id = "' + vars["google_default_client_id"] + '"\n')
+    #dargs.write('google_default_client_secret = "' + vars["google_default_client_secret"] + '"\n')
+    dargs.write('disable_fieldtrial_testing_config = true\n')
+    dargs.write('enable_nacl = false\n')
+    dargs.write('blink_symbol_level = 0\n')
+    dargs.write('symbol_level = 1\n')
+    dargs.write('is_official_build = true\n')
+    dargs.write('is_component_ffmpeg = true\n')
 
     dargs.close()
 
@@ -64,8 +80,9 @@ if __name__ == '__main__':
     gyp_vars["target_sysroot"] = os.path.abspath("build/linux/pios_bullseye_armhf-sysroot")
 
     docopy("armv7-bullseye", gyp_vars, is_ozone=True)
+    docopy("armv7-bullseye", gyp_vars, is_ozone=True, extra_codecs=False)
 
     gyp_vars["target_sysroot"] = os.path.abspath("build/linux/pios_bullseye_arm64-sysroot")
 
     docopy("arm64-bullseye", gyp_vars, is_ozone=True)
-
+    docopy("arm64-bullseye", gyp_vars, is_ozone=True, extra_codecs=False)
