--- a/build/config/linux/BUILD.gn
+++ b/build/config/linux/BUILD.gn
@@ -4,7 +4,6 @@
 
 import("//build/config/c++/c++.gni")
 import("//build/config/linux/pkg_config.gni")
-import("//build/config/sysroot.gni")
 import("//build/config/ui.gni")
 
 group("linux") {
@@ -113,14 +112,3 @@
 config("export_dynamic") {
   ldflags = [ "-rdynamic" ]
 }
-
-# MMAL ld flags
-config("use_mmal") {
-  if (is_linux && target_cpu == "arm") {
-    ldflags = [
-      "-L$sysroot/opt/vc/lib",
-      "-Wl,--rpath=/opt/vc/lib,--rpath-link=$sysroot/opt/vc/lib",
-      "-Wl,--start-group", "-lbcm_host", "-lmmal", "-lmmal_util", "-lmmal_core", "-lmmal_vc_client", "-lvcos", "-lvcsm", "-lvchostif", "-lvchiq_arm", "-Wl,--end-group",
-    ]
-  }
-}
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -28,6 +28,7 @@
 import("//tools/v8_context_snapshot/v8_context_snapshot.gni")
 import("//ui/gl/features.gni")
 import("//v8/gni/v8.gni")
+import("//build/config/sysroot.gni")
 
 if (is_android) {
   import("//build/config/android/rules.gni")
@@ -310,7 +311,11 @@
         # Add pi link stuff
         # * is there a better place?
         if (is_linux && target_cpu == "arm") {
-          configs += [ "//build/config/linux:use_mmal" ]
+          ldflags += [
+            "-L$sysroot/opt/vc/lib",
+            "-Wl,--rpath=/opt/vc/lib,--rpath-link=$sysroot/opt/vc/lib",
+            "-Wl,--start-group", "-lbcm_host", "-lmmal", "-lmmal_util", "-lmmal_core", "-lmmal_vc_client", "-lvcos", "-lvcsm", "-lvchostif", "-lvchiq_arm", "-Wl,--end-group",
+          ]
           if (symbol_level >= 2) {
             # For debug symbols!
             ldflags += [ "-Wl,--long-plt" ]
--- a/media/filters/BUILD.gn
+++ b/media/filters/BUILD.gn
@@ -133,7 +133,6 @@
       "$target_sysroot/opt/vc/include/interface/vmcs_host/linux",
     ]
 
-    public_configs = [ "//build/config/linux:use_mmal" ]
     public_deps += [ "//services/service_manager/zygote" ]
   }
 
