--- a/src/content/common/user_agent.cc
+++ b/src/content/common/user_agent.cc
@@ -198,7 +198,7 @@ std::string GetOSVersion(IncludeAndroidB
                       "%s%s", android_version_str.c_str(),
                       android_info_str.c_str()
 #else
-                      ""
+                      "13597.84.0"
 #endif
   );
   return os_version;
@@ -232,7 +232,7 @@ std::string BuildOSCpuInfoFromOSVersionA
   base::StringAppendF(&os_cpu,
 #if defined(OS_MAC)
                       "%s Mac OS X %s", cpu_type.c_str(), os_version.c_str()
-#elif BUILDFLAG(IS_CHROMEOS_ASH) || BUILDFLAG(IS_CHROMEOS_LACROS)
+#elif BUILDFLAG(IS_CHROMEOS_ASH) || BUILDFLAG(IS_CHROMEOS_LACROS) || defined(OS_LINUX)
                       "CrOS "
                       "%s %s",
                       cpu_type.c_str(),  // e.g. i686
