--- a/chrome/browser/notifications/notification_platform_bridge_linux.cc
+++ b/chrome/browser/notifications/notification_platform_bridge_linux.cc
@@ -224,7 +224,7 @@ bool ShouldMarkPersistentNotificationsAs
   // https://github.com/linuxmint/Cinnamon/issues/7179
   // For this reason, we mark a notification that should not time out as
   // critical unless we are on KDE Plasma which follows the notification spec.
-  return server_name != "Plasma";
+  return server_name != "Plasma" && server_name != "wf-panel-pi";
 }
 
 void ForwardNotificationOperation(NotificationOperation operation,
