--- a/src/gin/v8_initializer.cc
+++ b/src/gin/v8_initializer.cc
@@ -76,7 +76,11 @@ const char kSnapshotFileName32[] = "snap
 #endif
 
 #else  // defined(OS_ANDROID)
+#if __ARM_ARCH >= 7
+const char kV8ContextSnapshotFileName[] = "v8_context_snapshot_v7.bin";
+#else
 const char kV8ContextSnapshotFileName[] = "v8_context_snapshot.bin";
+#endif
 const char kSnapshotFileName[] = "snapshot_blob.bin";
 #endif  // defined(OS_ANDROID)
 
--- a/src/tools/v8_context_snapshot/BUILD.gn
+++ b/src/tools/v8_context_snapshot/BUILD.gn
@@ -12,6 +12,7 @@ import("//tools/v8_context_snapshot/v8_c
 import("//build/config/c++/c++.gni")
 import("//build/config/compiler/compiler.gni")
 import("//v8/gni/snapshot_toolchain.gni")
+import("//build/config/arm.gni")
 
 if (is_android) {
   import("//build/config/android/rules.gni")
@@ -49,7 +50,11 @@ if (use_v8_context_snapshot) {
 
   action("generate_v8_context_snapshot") {
     script = "//build/gn_run_binary.py"
-    output_file = "$root_out_dir/v8_context_snapshot.bin"
+    if (arm_use_neon) {
+      output_file = "$root_out_dir/v8_context_snapshot_v7.bin"
+    } else {
+      output_file = "$root_out_dir/v8_context_snapshot.bin"
+    }
     output_path = rebase_path(output_file, root_build_dir)
 
     args = [
