--- a/src/third_party/widevine/cdm/widevine.gni
+++ b/src/third_party/widevine/cdm/widevine.gni
@@ -23,7 +23,7 @@ if (is_chromeos && !is_chromeos_device)
 # supported via Android MediaDrm API.
 library_widevine_cdm_available =
     (is_chromeos && (target_cpu == "x64" || target_cpu == "arm")) ||
-    (is_desktop_linux && (target_cpu == "x86" || target_cpu == "x64")) ||
+    (is_desktop_linux)  ||
     (is_mac && (target_cpu == "x64" || target_cpu == "arm64")) ||
     (is_win && (target_cpu == "x86" || target_cpu == "x64"))
 
