--- a/src/third_party/widevine/cdm/widevine.gni
+++ b/src/third_party/widevine/cdm/widevine.gni
@@ -19,7 +19,7 @@ declare_args() {
 # TODO(hmchen): Provide support on "arm64".
 library_widevine_cdm_available =
     (is_chromeos && (target_cpu == "x64" || target_cpu == "arm")) ||
-    (is_desktop_linux && (target_cpu == "x86" || target_cpu == "x64")) ||
+    (is_desktop_linux && (target_cpu == "x86" || target_cpu == "x64" || target_cpu == "arm")) ||
     (is_mac && target_cpu == "x64") ||
     (is_win && (target_cpu == "x86" || target_cpu == "x64"))
 
