--- a/mojo/public/rust/mojom_value_parser/lib_core.rs
+++ b/mojo/public/rust/mojom_value_parser/lib_core.rs
@@ -10,6 +10,8 @@
 //! depend on this crate; instead, they should use the `mojom_value_parser`
 //! crate (rooted in `lib_pub.rs`), which only exports the `api` module and some
 //! fundamental types.
+#![feature(let_chains)]
+#![feature(const_type_id)]
 
 mod api;
 mod ast;
--- a/mojo/public/rust/system/lib.rs
+++ b/mojo/public/rust/system/lib.rs
@@ -1,6 +1,7 @@
 // Copyright 2025 The Chromium Authors
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
+#![feature(ptr_as_ref_unchecked)]
 
 pub mod data_pipe;
 pub mod message;
