Index: chromium-130.0.6723.44/sandbox/linux/services/credentials.cc
===================================================================
--- chromium-130.0.6723.44.orig/sandbox/linux/services/credentials.cc
+++ chromium-130.0.6723.44/sandbox/linux/services/credentials.cc
@@ -87,7 +87,7 @@ bool ChrootToSafeEmptyDir() {
   alignas(16) char stack_buf[PTHREAD_STACK_MIN_CONST];
 
 #if defined(ARCH_CPU_X86_FAMILY) || defined(ARCH_CPU_ARM_FAMILY) || \
-    defined(ARCH_CPU_MIPS_FAMILY)
+    defined(ARCH_CPU_MIPS_FAMILY) || defined(ARCH_CPU_PPC64_FAMILY)
   // The stack grows downward.
   void* stack = stack_buf + sizeof(stack_buf);
 #else
