Index: chromium-130.0.6723.44/build/config/compiler/BUILD.gn
===================================================================
--- chromium-130.0.6723.44.orig/build/config/compiler/BUILD.gn
+++ chromium-130.0.6723.44/build/config/compiler/BUILD.gn
@@ -1474,7 +1474,7 @@ config("compiler_cpu_abi") {
         cflags += [ "-maix64" ]
         ldflags += [ "-maix64" ]
       } else {
-        cflags += [ "-m64" ]
+        cflags += [ "-m64", "-mcpu=power9", "-mtune=power9" ]
         ldflags += [ "-m64" ]
       }
     } else if (current_cpu == "riscv64") {
Index: chromium-130.0.6723.44/third_party/libvpx/BUILD.gn
===================================================================
--- chromium-130.0.6723.44.orig/third_party/libvpx/BUILD.gn
+++ chromium-130.0.6723.44/third_party/libvpx/BUILD.gn
@@ -105,7 +105,7 @@ config("libvpx_config") {
 
   if (current_cpu == "ppc64") {
     cflags += [
-      "-mcpu=power8",
+      "-mcpu=power9",
       "-maltivec",
       "-mvsx",
     ]
Index: chromium-130.0.6723.44/v8/BUILD.gn
===================================================================
--- chromium-130.0.6723.44.orig/v8/BUILD.gn
+++ chromium-130.0.6723.44/v8/BUILD.gn
@@ -1434,7 +1434,7 @@ config("toolchain") {
     if (current_os == "linux") {
         cflags += [
         # Enable usage of AltiVec, VSX, and other POWER8 and higher features
-        "-mcpu=power8",
+        "-mcpu=power9",
         "-maltivec",
         "-mvsx",
       ]
