Index: chromium-130.0.6723.44/build/rust/cargo_crate.gni
===================================================================
--- chromium-130.0.6723.44.orig/build/rust/cargo_crate.gni
+++ chromium-130.0.6723.44/build/rust/cargo_crate.gni
@@ -430,7 +430,7 @@ template("cargo_crate") {
         ]
       }
       if (current_cpu == "arm64" || current_cpu == "x64" ||
-          current_cpu == "loong64" || current_cpu == "riscv64") {
+          current_cpu == "loong64" || current_cpu == "riscv64" || current_cpu == "ppc64") {
         args += [
           "--pointer-width",
           "64",
