Index: chromium-134.0.6998.35/build/rust/cargo_crate.gni
===================================================================
--- chromium-134.0.6998.35.orig/build/rust/cargo_crate.gni
+++ chromium-134.0.6998.35/build/rust/cargo_crate.gni
@@ -450,7 +450,7 @@ template("cargo_crate") {
         ]
       }
       if (current_cpu == "arm64" || current_cpu == "x64" ||
-          current_cpu == "loong64" || current_cpu == "riscv64") {
+          current_cpu == "loong64" || current_cpu == "riscv64" || current_cpu == "ppc64") {
         args += [
           "--pointer-width",
           "64",
