
Fix failure to find size_t definition

Index: chromium-116.0.5845.82/components/bookmarks/common/url_load_stats.h
===================================================================
--- chromium-116.0.5845.82.orig/components/bookmarks/common/url_load_stats.h
+++ chromium-116.0.5845.82/components/bookmarks/common/url_load_stats.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_BOOKMARKS_COMMON_URL_LOAD_STATS_H_
 #define COMPONENTS_BOOKMARKS_COMMON_URL_LOAD_STATS_H_
 
+#include <cstddef>
 #include <cstdint>
 #include <vector>
 
