Index: chromium-130.0.6723.44/base/compiler_specific.h
===================================================================
--- chromium-130.0.6723.44.orig/base/compiler_specific.h
+++ chromium-130.0.6723.44/base/compiler_specific.h
@@ -88,7 +88,7 @@
 // Can be used only on return statements, even for functions returning void.
 // Caller and callee must have the same number of arguments and its types must
 // be "similar".
-#if defined(__clang__) && HAS_ATTRIBUTE(musttail)
+#if defined(__clang__) && HAS_ATTRIBUTE(musttail) && !defined(__powerpc64__)
 #define MUSTTAIL [[clang::musttail]]
 #else
 #define MUSTTAIL
