Index: chromium-127.0.6533.72/third_party/skia/src/core/SkRasterPipeline.h
===================================================================
--- chromium-127.0.6533.72.orig/third_party/skia/src/core/SkRasterPipeline.h
+++ chromium-127.0.6533.72/third_party/skia/src/core/SkRasterPipeline.h
@@ -27,7 +27,7 @@ struct SkImageInfo;
 struct skcms_TransferFunction;
 
 #if __has_cpp_attribute(clang::musttail) && !defined(__EMSCRIPTEN__) && !defined(SK_CPU_ARM32) && \
-        !defined(SK_CPU_LOONGARCH)
+        !defined(SK_CPU_LOONGARCH) && !defined(__powerpc64__)
     #define SK_HAS_MUSTTAIL 1
 #else
     #define SK_HAS_MUSTTAIL 0
