--- a/third_party/node/node_modules.py
+++ b/third_party/node/node_modules.py
@@ -14,11 +14,13 @@ def _path_in_node_modules(*args):
 # Typescript is not approved for general use in Chromium. Email chromium-dev
 # if you want to use it.
 def PathToTypescript():
-  return '/usr/bin/tsc'
+  return _path_in_node_modules('typescript', 'bin', 'tsc')
+  #return '/usr/bin/tsc'
 
 def PathToRollup():
   return '/usr/share/nodejs/rollup/dist/bin/rollup'
 
+
 def PathToPolymerCssBuild():
   return _path_in_node_modules('polymer-css-build', 'bin', 'polymer-css-build')
 
